menubot/index.js

```diff
@@ -0,0 +1,127 @@
+
+  // too spammy but good for debugging
+  console.log(`Processing tickInfo: ${JSON.stringify(tickInfo)}`)
+
+  switch (sceneName) {
+    case "MainMenu":
+
+      const hostButton = getInteractableButton(tickInfo, "RGHostButton");
+      if (hostButton && stateFlags[0] && !stateFlags[1]) {
+        clickButton(hostButton.id, actionQueue);
+        stateFlags[1] = true
+      }
+
+      const startButton = getInteractableButton(tickInfo, "StartWithRGButton");
+      if (startButton && stateFlags[5] && !stateFlags[0]) {
+        clickButton(startButton.id, actionQueue);
+        stateFlags[0] = true
+      }
+
+      const selectProfileButton = getInteractableButton(tickInfo, "SelectProfileButton");
+      if (selectProfileButton && stateFlags[4] && !stateFlags[5]) {
+        clickButton(selectProfileButton.id, actionQueue);
+        stateFlags[5] = true
+      }
+
+      const profileMenuButton = getInteractableButton(tickInfo, "ProfileMenuButton");
+      if (profileMenuButton && !stateFlags[4]) {
+        clickButton(profileMenuButton.id, actionQueue);
+        stateFlags[4] = true
+      }
+
+      break;
+    case "CharSelect":
+      const readyButton = getInteractableButton(tickInfo, "ReadyButton");
+      if (readyButton && stateFlags[2] && !stateFlags[3]) {
+        clickButton(readyButton.id, actionQueue);
+        stateFlags[3] = true
+      }
+
+      const seat7Button = getInteractableButton(tickInfo, "Seat7Button");
+      if (seat7Button && !stateFlags[2]) {
+        clickButton(seat7Button.id, actionQueue);
+        stateFlags[2] = true
+      }
+
+      break;
+    default:
+      // teardown myself
+      console.log(`Game started, bot is Complete`)
+      botComplete = true;
+      break;
+  }
+
+}
+
+function getInteractableButton(tickInfo, buttonName) {
+  const buttons = RGBot.getEntitiesOfType(tickInfo, buttonName);
+  //console.log(`Found buttons: ${JSON.stringify(buttons)}`)
+  if (buttons && buttons.length > 0) {
+    const button = buttons[0];
+    if (button.interactable) {
+      //console.log(`Returning button: ${JSON.stringify(button)}`)
+      return button;
+    }
+  }
+  return null;
+}
+
+function clickButton(targetId, actionQueue) {
+  console.log(`Clicking button: ${targetId}`)
+  const input = {
+    targetId: targetId
+  }
+  actionQueue.queue("ClickButton", input)
+}
+
+/**
+ * Defines the type of character that the game should use for this bot.
+ */
+export function getCharacterType() {
+  return CharInfo.type[charType];
+}
+
+/**
+ * Let server know if i have finished my processing
+ * @returns {boolean} true if done processing and ready to be torn down
+ */
+export function isComplete() {
+  return botComplete;
+}
+
+/**
+    One of ...
+    SPAWNABLE - Server disconnects/ends bot on match/game-scene teardown
+    PERSISTENT - Bot is responsible for disconnecting / ending itself
+ */
+export function getBotLifecycle() {
+  return 'PERSISTENT';
+}
+
+export function isSpawnable() {
```

Comment:
During your demo I mentioned that including these settings on the bot itself rather than requiring their inclusion in the botcode could offer a better user experience. You made some good points around how we expect someone to set these on the bot and making sure they have a way to update them afterwards without re-creating their bot. I think it makes sense to accept this as-is and make a not to revisit this idea as part of [REG-1011](https://linear.app/regression-games/issue/REG-1011/revisit-and-clean-up-the-design-of-the-unity-js-sdk-to-be-more-user)